monitoring/README.md

```diff
@@ -0,0 +1,185 @@
+Also, you can find examples of the Traces in the [examples](#examples).
+
+## Spans
+
+A **Span** represents a unit of work or operation. Spans are the building blocks of Trace. In OpenTelemetry, they include the following information:
+
+- Name
+- Parent span ID (empty for root spans)
+- Start and End Timestamps
+- [Span Context](https://opentelemetry.io/docs/concepts/signals/traces/#span-context)
+- [Attributes](https://opentelemetry.io/docs/concepts/signals/traces/#attributes)
+- [Span Events](https://opentelemetry.io/docs/concepts/signals/traces/#span-events)
+- [Span Links](https://opentelemetry.io/docs/concepts/signals/traces/#span-links)
+- [Span Status](https://opentelemetry.io/docs/concepts/signals/traces/#span-status)
+
+Spans can be nested, as is implied by the presence of a parent span ID: child spans represent sub-operations. This allows spans to more accurately capture the work done in an application.
+
+Spans follow the [Open Telemetry Semantic Conventions](https://opentelemetry.io/docs/concepts/semantic-conventions/). That ensures correlation on naming between spans and [logs](../logger/).
+
+The naming conversions can be found in [internal/config/commontags.go](../internal/config/commontags.go).
+
+Creating a Span it requires a Tracer. The library provides a simple way to create Spans. Furthermore, exposing the Span interface you can easily add Attributes, Events, Links, Status and Record Errors in your Spans.
+
+On top of the Span interface from Otel, the library also exposes two more methods on the Span, `EndWithError` and `EndSuccessfully`.
+
+By default the Span interface from Otel exposes only the method `End` to end the Span.
+The new method `EndWithError` is also ending the Span, but it will flag the Span as errored and will also include the error into the Span Events (as it must be based on Otel). On the other hand, `EndSuccessfully` ends the Span and updates the Status as `Ok`.
+
+> [!IMPORTANT]
+> You must **always** close the Spans, otherwise you might experience OOM-kills in your services. One way to always ensure the spans are closing, you can use the [go-spancheck](https://github.com/jjti/go-spancheck) rule in the [golangci-lint](https://github.com/golangci/golangci-lint).
+
+### Automatic Correlation
+
+In order for the library to give a better experience when working with Traces, Spans and Logs, it also provides some automations to make the usage easier.
+
+When you create a new Span, it returns back a new `context.Context` value. This new context contains the Span information.
+
+When logging inside a block of code that is wrapped in a Span, it is recommended to pass to the Logger the new context that includes the Span information. That won't only inluce the Trace and Span IDs into the log, but also will automatically include into the Spans the attributes that are being passed to the Logger.
+
+That ensures useful information for debugging will be present at the same time in both the logs and the spans.
```

Comment:
`That ensures useful information for debugging will be present in both the logs and the spans.`